Python/pythonrun.c

```diff
@@ -188,6 +188,25 @@ PyRun_InteractiveLoopFlags(FILE *fp, const char *filename, PyCompilerFlags *flag
 
 }
 
+static int
+PyRun_IsInteractiveExitCommand(mod_ty mod) {
+    if (asdl_seq_LEN(mod->v.Module.body) != 1) {
+        return 0;
+    }
+    stmt_ty statement = asdl_seq_GET(mod->v.Module.body, 0);
+    if (statement->kind != Expr_kind) {
+        return 0;
+    }
+    expr_ty expr = statement->v.Expr.value;
+    if (expr->kind != Name_kind) {
+        return 0;
+    }
+    if (expr->v.Name.ctx != Load) {
+        return 0;
+    }
+    return PyUnicode_CompareWithASCIIString(expr->v.Name.id, "exit") == 0 ||
+           PyUnicode_CompareWithASCIIString(expr->v.Name.id, "quit") == 0;
+}
```

Comment:
Tests of the object associated with the names is needed to not turn them into semi-keywords, as in the the following.
```
>>> exit = 3
>>> exit

f:\dev\3x>
```